mac/hid.c

```diff
@@ -696,11 +792,11 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 	io_registry_entry_t entry = MACH_PORT_NULL;
 	IOReturn ret = kIOReturnInvalid;
 
+	dev = new_hid_device();
+
 	/* Set up the HID Manager if it hasn't been done */
 	if (hid_init() < 0)
-		goto return_error;
-
-	dev = new_hid_device();
+		return NULL;
```

Comment:
this is a regression - the changes applied not on top of the latest master
see: https://github.com/libusb/hidapi/commit/d8b6588d09885a9eafce995a44b731f544d1c234

I'll check carefully full PR a bit later

Reply:
I think all of the code below shouldn't be changed at all

Reply:
same applies to changes in get/set_input/feature_report